en/docs/design/api-security/opa-validation/overview.md

```diff
@@ -1,33 +1,55 @@
 # Validate Request with Open Policy Agent (OPA)
 
-The [Open Policy Agent (OPA)](https://openpolicyagent.org/) is an open source, general-purpose policy engine that unifies policy enforcement. In Choreo Connect, you can offload some responsibility of making the decision to authorize or not when a consumer invokes APIs based on policies attached to APIs. Both WSO2 API Manager and Choreo Connect supports Open Policy Agent based request validation.
+The [Open Policy Agent (OPA)](https://openpolicyagent.org/) is an open source, general-purpose policy engine that unifies policy enforcement. 
+With the ability to define complex policies as a code, OPA provides the much needed flexibility over standard static access control mechanisms such as role based or attribute based access control.
 
-## Attaching OPA Policy
+You can offload some responsibility of making the decision to authorize or not when a consumer invokes APIs based on policies attached to APIs. Both WSO2 API Manager regular Gateway and Choreo Connect supports Open Policy Agent based request validation. Open Policy Agent comprises of a policy engine that allows users (or systems) to query policies for decisions. 
+A policy for Open Policy Agent can be though as a set of rules and policy engine evaluate these rules based on the input it receives.
 
-You can attach an OPA policy to the API that you want to enforce OPA validation. In the publisher portal, select the API that you want to configure the OPA policy and navigate to **Develop -> API Configurations -> Policies**.
+WSO2 API Manager uses this unique policy engine to evaluate the requests that the Gateway receives and decides the fate of the request.
 
-Select the gateway that the API required to be deployed in the upper right corner of the page.
+<a href="{{base_path}}/assets/img/design/security/opa/opa-policy-architecture.png">
+    <img src="{{base_path}}/assets/img/design/security/opa/opa-policy-architecture.png" alt="OPA Architecture"/>
+</a>
+
+## How to use OPA for request validation?
+
+In oder to evaluate requests with OPA, first you need to deploy OPA as a host-level daemon or a sidecar container. Once a Gateway receives a request,
+it extracts a predefined set of meta-data from the request and query OPA via HTTP/HTTPS. The OPA Policy engine evaluates the request meta-data against the configured policy and returns its validation response.
+The Gateway allows or blocks the request based on the response.
+
+!!! note
+    Gateway node should be able to communicate with this OPA server. It can be either HTTP or HTTPS and if HTTPS is used, make sure that you upload the certificates of the OPA server to the client trust store of the gateway.
```

Comment:
```suggestion
    The Gateway node should be able to communicate with this OPA server. It can be either HTTP or HTTPS, and if HTTPS is used, make sure that you upload the certificates of the OPA server to the client trust store of the Gateway.
```